### PR TITLE
sprint-automation: add a role digest for the following week

### PR DESCRIPTION
**Pull request by smg247**
This adds the role digest mentioned in [DPTP-2432](https://issues.redhat.com/browse/DPTP-2432).
It also adds an easy way to test using our testing slack, and a README describing `sprint-automation`.
I still want to follow up with an "on demand" view of upcoming on call role's, but that will use the `slack-bot` and be more complex.

Next up:
add a new job to `infra-periodics.yaml` that runs weekly on mondays using the new `week-start` flag. Also, will need to stop the current `periodic-sprint-automation` from running on mondays. I think the work I've done here will make it easier to fix [DPTP-2403](https://issues.redhat.com/browse/DPTP-2403) as well, by allowing us to query for active roles as of noon UTC on mondays.

### CODE REVIEW

File: cmd/sprint-automation/main.go [hunk 1]
```diff
@@ -303,6 +318,73 @@ func userOnCallDuring(client *pagerduty.Client, query string, since, until time.
...
 
+func sendNextWeeksRoleDigest(client *pagerduty.Client, slackClient *slack.Client) error {
+	var errors []error
+	// Use one week from now at noon UTC to ensure that PD roles have begun
+	nextWeek := time.Now().Add(7 * 24 * time.Hour)
+	nextWeek = time.Date(nextWeek.Year(), nextWeek.Month(), nextWeek.Day(), 12, 0, 0, 0, time.UTC)
+
+	userIdsByRole, errs := usersOnCallAtTime(client, slackClient, nextWeek)
+	if len(errs) > 0 {
+		errors = append(errors, errs...)
```
[the remark this answers is not in the document]
Comment (smg247) [reply cmd/sprint-automation/main.go:329]: We can still go on. It is possible that one role is not found (or one user) while the others are all fine, and something is better than nothing. Good note about the log error though, I will add that.

File: hack/local-sprint-automation.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	oc --context app.ci --namespace ci --as system:admin "$@"
+}
+
+os::log::info "Extracting production data we need to run sprint-automation..."
+OC extract secret/slack-credentials-dptp-bot-alpha --keys oauth_token --to "${data}"
+OC extract secret/jira-token-dptp-bot --keys token --to "${data}/jira"
+OC extract secret/pagerduty --keys token --to "${data}/pd"
+
+os::log::info "Running sprint-automation"
+sprint-automation --week-start=true --slack-token-path "${data}/oauth_token" --pager-duty-token-file="${data}/pd/token" --jira-bearer-token-file="${data}/jira/token" --jira-endpoint=https://issues.redhat.com --log-level=trace
```
[the remark this answers is not in the document]
Comment (smg247) [reply hack/local-sprint-automation.sh:32]: good call!

**Review by hongkailiu — APPROVED** [review]

**Pull request merged**